synthesizer/src/vm/finalize.rs

```diff
@@ -2161,6 +2183,25 @@ finalize compute:
             assert!(expected_bonded.contains(entry));
         }
 
+        // Get and check the withdraw mapping.
+        let actual_withdraw = vm.finalize_store().get_mapping_confirmed(program_id, withdraw_mapping_name).unwrap();
+        let expected_withdraw = bonded_balances
+            .iter()
+            .map(|(address, (_, withdrawal_address, _))| {
+                (
+                    Plaintext::from_str(&address.to_string()).unwrap(),
+                    Value::from_str(&withdrawal_address.to_string()).unwrap(),
+                )
+            })
+            .collect_vec();
+        // Note that `actual_withdraw` and `expected_withdraw` are vectors and not necessarily in the same order.
+        // By checking that the lengths of the vector are equal and that all entries in `actual_withdraw` are in `expected_withdraw`,
+        // we can ensure that the two vectors contain the same data.
+        assert_eq!(actual_withdraw.len(), expected_withdraw.len());
+        for entry in actual_withdraw.iter() {
+            assert!(expected_withdraw.contains(entry));
```

Comment:
a linear search could become slow if the `expected_withdraw` vec is large :thinking:

Reply:
ah, but it's just a test; nvm then